src/miragevpn_server.md

```diff
@@ -0,0 +1,125 @@
+The MirageVPN server is a unikernel which listens on a port for client
+connections, and provides Internet connectivity (with NAT) for all
+authenticated clients. A single network interface is used for both client
+connections and providing connectivity.
+
+The unikernel will encrypt all packets received from the Internet and send
+them to the respective client. All packets received from a client will be
+decrypted and forwarded to the Internet. If "client-to-client" is enabled
+in the configuration, packets from one client which destination is another
+client will be forwarded by the server.
+
+# Scope of the server
+
+The scope is at the moment limited to IPv4 (no IPv6), the server only listens
+on TCP (no UDP). Only layer 3 networking is supported (tun device), there's no
+support for tap devices.
+
+The server will route all traffic to its default gateway. If "client-to-client"
+is specified, packets for other clients will be forwarded to the specific
+client.
+
+Only a single network interface is used by the server, where both TCP listening
+and forwarding packets (using NAT) is done. NAT will always be used.
+
+## Authentication
+
+The server as is only authenticates via X.509 certificates. It is possible to
+authenticate username and password, but since we do can not in a MirageOS
+unikernel execute shell scripts, the verification hook script won't work.
+If you need username and password authentication, please get in touch via our
+issue tracker and we will find a solution.
+
+# Producing the unikernel binary
+
+We will provide reproducible binaries in the future, here we document how to
+build the unikernel from source.
+
+## Prerequisites
+
+First, make sure to have ["opam"](https://opam.ocaml.org) and
+["mirage"](https://mirage.io) installed on your system.
+
+## Git repository
+
+Do a `git clone https://github.com/robur-coop/miragevpn.git` to retrieve the
+source code of the MirageVPN server.
+
+## Building
+
+Inside of the cloned repository, execute `mirage configure` (other targets are
```

Comment:
Add a reference to mirage.io on how to install mirage maybe? Or we preface with a prerequisite to have mirage installed

Reply:
There's a "prerequisite" which says "opam and mirage" need to be present -- with links to opam.ocaml.org and mirage.io. Is that sufficient?

Reply:
Ah sorry yes that is fine and sufficient!